torch/_functorch/_aot_autograd/traced_function_transforms.py

```diff
@@ -350,12 +349,28 @@ def _functionalized_f_helper(*args):
         disable_above = torch._C._ExcludeDispatchKeyGuard(
             torch._C.DispatchKeySet(torch._C.DispatchKey.Functionalize)
         )
-        with disable_above, FunctionalTensorMode(aot_config.pre_dispatch):
+
+        with disable_above:
             # Wrap inputs into functional wrappers
             f_args = pytree.tree_map(to_fun, args)
+
+            # Populate the current FunctionalTensorMode with the tokens per
```

Comment:
Maybe I am too big a fan of `Note [...]` comments...

Do you mind adding a comment like:
```
Note [FunctionalTensorMode is Stateful]
```

At the top of FunctionalTensorMode that briefly explains why it's stateful, and then link the comment here? (aka the one spot in the codebase today that needs to handle propagating the state)